packages/plugin-list/List.ts

```diff
@@ -102,74 +101,66 @@ export class List extends JWPlugin {
             // are rendered *as* "LI" nodes.
-            const listItem = currentContext.currentNode;
-            const domListNode = currentContext.parentNode;
             // Check if previous "LI" can be reused or create a new one.
-            let liNode: Element;
-            if (listItem.is(ListNode) && domListNode.childNodes.length) {
+            let domListItem: Element;
+            if (node.is(ListNode) && domParent.childNodes.length) {
                 // Render an indented list in the list item that precedes it.
                 // eg.: <ul><li>title: <ul><li>indented</li></ul></ul>
-                liNode = domListNode.childNodes[domListNode.childNodes.length - 1] as Element;
+                domListItem = domParent.childNodes[domParent.childNodes.length - 1] as Element;
             } else {
-                liNode = document.createElement('li');
+                domListItem = document.createElement('li');
             }
-            domListNode.appendChild(liNode);
+            domParent.appendChild(domListItem);
 
-            const renderingMap: DomRenderingMap = new Map();
-            currentContext.parentNode = liNode;
             // Direct ListNode's VElement children "P" are rendered as "LI"
             // while other nodes will be rendered inside the "LI".
-            if (listItem.is(VElement) && listItem.htmlTag === 'P') {
-                // Mark the "P" as rendered by the "LI".
-                renderingMap.set(liNode, [listItem]);
-                // TODO: this should be generic.
-                if (!listItem.hasChildren()) {
-                    liNode.appendChild(document.createElement('BR'));
-                }
+            if (node.is(VElement) && node.htmlTag === 'P') {
+                // Render the node's children
+                let rendered = [];
+                node.children.forEach(child => {
+                    if (!rendered.includes(child)) {
+                        rendered = [
+                            ...rendered,
+                            ...this.editor.renderers.dom.renderNode(child, domListItem).keys(),
+                        ];
+                    }
+                });
             } else {
                 // The node was wrapped in a "LI" but needs to be rendered as well.
-                currentContext = this.editor.renderers.dom.renderNode({
-                    ...currentContext,
-                }) as DomRenderingContext;
+                this.editor.renderers.dom.renderNode(node, domListItem) as Map<VNode, Node[]>;
             }
-            this._isRenderingLi = false;
-            return [{ ...currentContext }, renderingMap];
+
+            const domNodes = [domListItem];
+            // TODO: this should be generic.
+            if (!node.hasChildren()) {
+                const br = document.createElement('BR');
+                domListItem.appendChild(br);
+                domNodes.push(br);
+            }
```

Comment:
In my understanding, the one and only time where you would still have to do that is in the case where you render a `P` as a `LI` because it is a child of a list and that particular `P` is empty. It's the only time where you "prevent" somewhat the node from rendering itself using the standard mechanism.

Reply:
So you want me to move this into the above `if`?